### PR TITLE
feat: add support for arbitrary foreign messages

### LINKED ISSUE #230: fix: add support for arbitrary foreign messages
Add `foreign` property to the receiver's declaration. If `true` then messages are considered as _foreign_, that is, having an arbitrary format and must be converted to UI message `{ payload: any }`.

Default to `false`, unless the `binding` property is implicitly defined.

## Change Requests

- [x] #232
- [x] feat(boot): add `foreign` property to the [receiver's constructor](https://github.com/toa-io/toa/blob/90d83aa206ce5f4c394135b4c747c999c058e532/runtime/boot/src/receivers.js#L16)
- [x] feat(core): add support for arbitrary foreign messages to the [receiver](https://github.com/toa-io/toa/blob/main/runtime/core/src/receiver.js)

### PR DESCRIPTION
**Pull request by burlakilia**
Close #230

### CODE REVIEW

File: runtime/norm/features/steps/parameters.js [hunk 1]
```diff
@@ -7,3 +7,9 @@ defineParameterType({
   regexp: /transition|observation|assignment/,
   transformer: (type) => type
 })
+
+defineParameterType({
```
Comment (temich) [inline runtime/norm/features/steps/parameters.js:11]: replace with `token`

File: runtime/norm/src/.component/.normalize/receivers.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+const receivers = (component) => {
+  if (component.receivers === undefined) return
+
+  for (const [_, receiver] of Object.entries(component.receivers)) {
```
Comment (temich) [inline runtime/norm/src/.component/.normalize/receivers.js:6]: Object.values

File: runtime/norm/src/.component/.normalize/receivers.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    if (receiver.binding !== undefined && !receiver.hasOwnProperty('foreign')) receiver.foreign = true
```
Comment (temich) [inline runtime/norm/src/.component/.normalize/receivers.js:7]: && receiver.foreign === undefined

Comment (temich) [reply runtime/norm/src/.component/.normalize/receivers.js:7]: And swap conditions

Comment (temich) [reply runtime/norm/src/.component/.normalize/receivers.js:7]:
```javascript
if (receiver.foreign === undefined) receiver.foreign = receiver.binding !== undefined
```

File: runtime/norm/src/.component/schema.yaml [hunk 4]
```diff
@@ -213,6 +213,9 @@ properties:
           adaptive:
             type: boolean
             default: false
+          foreign:
+            type: boolean
+            default: false
         required: [transition, bridge, path]
```
Comment (temich) [inline runtime/norm/src/.component/schema.yaml:219]: Add `foreign` to required

**Comment by temich** [comment]
Renamed to whole feature, added reference to #230 

Please update PR with implementation of other issues of this feature.

**Review by temich — CHANGES_REQUESTED** [review]

File: runtime/norm/features/steps/parameters.js (hunk 1 above)
Comment (burlakilia) [reply runtime/norm/features/steps/parameters.js:11]: use lable

File: runtime/norm/src/.component/.normalize/receivers.js (hunk 2 above)
Comment (burlakilia) [reply runtime/norm/src/.component/.normalize/receivers.js]: done

**Review by temich — APPROVED** [review]

**Pull request merged**